### PR TITLE
fix(config): centralize VULTISIG_CONFIG_DIR resolution

### LINKED ISSUE #1166: config: VULTISIG_CONFIG_DIR resolution is duplicated across CLI/shared consumers and already drifted
## Summary
`VULTISIG_CONFIG_DIR` resolution was duplicated across CLI/shared consumers and had already drifted on empty/whitespace handling. One CLI path treated `''` / `'   '` as unset, while adjacent config/credential/token-cache/journal paths still had their own copies.

## Evidence
- `clients/cli/src/lib/config.ts:44-49` canonicalized empty/whitespace handling
- stale copies remained in:
  - `packages/client-shared/src/credential-store.ts:15-17,89-91`
  - `packages/client-shared/src/config-store.ts:8-15`
  - `clients/cli/src/agent/session.ts:1327-1329`
  - `clients/cli/src/agent/broadcastJournal.ts:202-209`

## Impact
- config registry, credentials, token cache, journal, and vault storage can disagree about where state lives
- the same class of bug can recur anywhere a new consumer copies the rule instead of importing it

## Fix sketch
Extract one shared `getVultisigConfigDir()` helper (trim, treat empty/whitespace as unset, default to `~/.vultisig`) and route all CLI/shared stateful surfaces through it. Add co-location tests.

## Effort
Small

## Report
https://gist.github.com/gomesalexandre/fac3c1408ccefc7e90a16955b11e855b

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Summary
- extract a shared `getVultisigConfigDir()` helper in `@vultisig/client-shared`
- route config store, credential store, token cache, broadcast journal, and CLI config through it
- add regression tests for whitespace/empty overrides and cross-surface co-location

## Test plan
- [x] `yarn build:all`
- [x] `yarn lint`
- [x] `yarn test`
- [x] `yarn vitest run --config .config/vitest.cli.config.ts clients/cli/src/lib/__tests__/config.test.ts clients/cli/src/agent/__tests__/broadcastJournal.test.ts clients/cli/src/agent/__tests__/configPaths.test.ts`
- [x] `cd clients/cli && yarn vitest run tests/config-store.test.ts`
- [x] `cd clients/cli && VULTISIG_CONFIG_DIR='   ' node dist/index.js auth status`

## Receipts
- full workspace gates passed: `yarn build:all`, `yarn lint`, `yarn test`
- targeted CLI tests passed: 37 tests across config/journal/path-policy + 13 config-store tests
- runtime proof: with `VULTISIG_CONFIG_DIR='   '`, `node dist/index.js auth status` returned a successful JSON payload instead of resolving `''` / crashing on bad paths

Closes #1166

## Round report
- https://gist.github.com/gomesalexandre/fac3c1408ccefc7e90a16955b11e855b

## Summary by CodeRabbit

* **Bug Fixes**
  * Empty or whitespace-only `VULTISIG_CONFIG_DIR` values now reliably fall back to the default `.vultisig` directory.
  * Configuration, credentials, token cache, and broadcast journal paths now resolve consistently.
  * Explicit broadcast journal path overrides continue to take precedence.

* **Refactor**
  * Centralized configuration-directory resolution for more consistent behavior across the CLI and shared components.

### CODE REVIEW

File: packages/client-shared/src/config-dir.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { homedir } from 'node:os'
+import { join } from 'node:path'
+
+export function getVultisigConfigDir(): string {
+  const override = process.env.VULTISIG_CONFIG_DIR?.trim()
```
Comment (NeOMakinG) [inline packages/client-shared/src/config-dir.ts:5]: `should-fix:` trim() silently migrates a space-padded VULTISIG_CONFIG_DIR value, not just whitespace-only overrides — the changeset frames this as "treat empty/whitespace-only overrides as unset", but the helper also `.trim()`s any non-empty value. before this PR the token cache (`session.ts` used `??`, no trim) and the config/credential stores (`||`, no trim) wrote to the LITERAL spaced dir; e.g. `VULTISIG_CONFIG_DIR=' /vaults '` resolved credentials.enc under `' /vaults '`. after this PR they resolve under `/vaults`. that is a second silent relocation of the credential + vault-registry path, low-probability but undocumented and uncovered by tests.

fix: either keep the trim and call out the space-trimming relocation explicitly in the changeset (it touches credentials.enc + config.json paths), or add a regression test pinning a padded value to the trimmed dir so the behavior is intentional-and-covered rather than incidental.

**Review by NeOMakinG — COMMENTED** [review]
gm — multi-lane pass: claude + codex gpt-5.4. conceptual-only (no runtime this pass). comment-only.

## should-fix
### 1. trim() silently migrates a space-padded VULTISIG_CONFIG_DIR value, not just whitespace-only overrides — `packages/client-shared/src/config-dir.ts:5`
the changeset frames this as "treat empty/whitespace-only overrides as unset", but the helper also `.trim()`s any non-empty value. before this PR the token cache (`session.ts` used `??`, no trim) and the config/credential stores (`||`, no trim) wrote to the LITERAL spaced dir; e.g. `VULTISIG_CONFIG_DIR=' /vaults '` resolved credentials.enc under `' /vaults '`. after this PR they resolve under `/vaults`. that is a second silent relocation of the credential + vault-registry path, low-probability but undocumented and uncovered by tests.
action: either keep the trim and call out the space-trimming relocation explicitly in the changeset (it touches credentials.enc + config.json paths), or add a regression test pinning a padded value to the trimmed dir so the behavior is intentional-and-covered rather than incidental.

## suggestion
### 1. one raw untrimmed VULTISIG_CONFIG_DIR reader left outside the shared resolver — `clients/cli/src/lib/config.ts:199`
`EnvVars.getConfigDir()` still returns `process.env.VULTISIG_CONFIG_DIR` raw (no trim, undefined-if-unset). it currently has zero consumers, so it is inert today — but it is a latent trap: the moment anything builds a path from `EnvVars.getConfigDir()` it resurrects the exact empty/whitespace bug this PR centralizes away.
action: route it through `getVultisigConfigDir()` or delete the dead accessor so the centralization is airtight.

## q
- confirm the whitespace-normalization is a deliberate behavior change: pre-PR `config-store`/`credential-store` used `||` (whitespace `'   '` passed through as truthy) and `session.ts` used `??` (empty `''` passed through) — all now normalize to `~/.vultisig`. reads as an intentional bugfix; flagging only so it is a conscious call, not a silent side-effect of the refactor.

## risk
low for the common case. verified locally: for the 99.9% path (env unset, or a clean absolute path set) resolution is byte-identical to the old per-call-site logic — `~/.vultisig` or the exact override, no vault/credential relocation. pure extraction: the new `getVultisigConfigDir()` body is character-identical to the old `lib/config.ts` inline resolver. tests proven non-inert: reverted the `.trim()` in config-dir.ts → the whitespace-override tests in `config-store.test.ts` + `configPaths.test.ts` go RED (2 failed); restored → 16/16 green. draft-hold: not approving a draft.

## verdict
COMMENTED

🤖 reviewed via [Claude Code](https://claude.com/claude-code) with codex `gpt-5.4-codex` conceptual pass.

**Review by rcoderdev — APPROVED** [review]
